apps/staking/app/register/[nodeId]/multi/SubmitMultiTab.tsx

```diff
@@ -0,0 +1,740 @@
+        }
+      },
+      gcTime: Number.POSITIVE_INFINITY,
+      staleTime: Number.POSITIVE_INFINITY,
+      refetchIntervalInBackground: true,
+      refetchOnWindowFocus: false,
+      refetchOnMount: false,
+    }
+  );
+
+  const contract = useMemo(() => getNonFinalizedLatestDeployedContributorContract(data), [data]);
+
+  useEffect(() => {
+    if (contract) {
+      setContract(contract);
+    } else {
+      setContract(null);
+    }
+  }, [contract]);
+
+  useEffect(() => {
+    if (data && 'network' in data && data.network) {
+      setNetworkInfo(data.network);
+    }
+  }, [data]);
+
+  /**
+   * Contract address is null if the contract is finalized or if there is no contract yet.
+   * In both cases we want to create a new contract.
+   */
+  const contractAddress = contract?.address ?? null;
+
+  const isContractStakedToByOperator = useMemo(() => {
+    if (!contract) return false;
+    return contract.contributors.some((contributor) => areHexesEqual(contributor.address, address));
+  }, [contract, address]);
+
+  const isContractStakeable = !!(
+    contract && contract.status === CONTRIBUTION_CONTRACT_STATUS.WaitForOperatorContrib
+  );
+
+  const stakingParamsMemoized = useMemo(() => {
+    return {
+      ...stakingParams,
+      contractAddress,
+    };
+  }, [stakingParams, contractAddress]);
+
+  const {
+    contributeStake,
+    resetContributeStake,
+    enabled: isContributeStakeEnabled,
+    allowanceReadStatus,
+    approveWriteStatus,
+    contributeFundsStatus,
+    approveErrorMessage,
+    contributeFundsErrorMessage,
+    approveWriteError,
+    approveSimulateError,
+    approveTransactionError,
+    contributeFundsSimulateError,
+    contributeFundsWriteError,
+    contributeFundsTransactionError,
+  } = useContributeStakeToOpenNode(stakingParamsMemoized);
+
+  /**
+   * If an error is thrown by this function that error is caught by the ErrorBoundary and handled there.
+   * If no error is thrown we treat the error as "recoverable", allowing a retry.
+   *
+   * Errors should be thrown if there is no way for the user to recover from the error, or no way
+   * to recover from the error programmatically.
+   */
+  const handleError = () => {
+    setIsSubmitting(false);
+
+    const contractError =
+      contributeFundsTransactionError ??
+      contributeFundsWriteError ??
+      contributeFundsSimulateError ??
+      approveTransactionError ??
+      approveWriteError ??
+      approveSimulateError ??
+      contractTransactionError ??
+      contractWriteError ??
+      contractSimulateError;
+
+    /**
+     * If there is a contract error, and it isn't explicitly recoverable, it will be thrown as unrecoverable.
+     */
+    if (contractError) {
+      const name = getContractErrorName(contractError);
+
+      if (recoverableErrors.has(name)) {
+        return;
+      }
+
+      setIsError(true);
+      throw contractError;
+    }
+
+    if (errorFetchContractDetails) {
+      // TODO: handle backend error
```

Comment:
ETA?

Reply:
:shrug:

Reply:
atm we dont have backend errors, so we just throw